src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -335,24 +380,23 @@ void RaftReplService::fetch_pending_data() {
     }
 }
 
-#if 0
 void RaftReplService::gc_repl_devs() {
     std::unique_lock lg(m_rd_map_mtx);
-    for (it = m_rd_map.begin(); it != m_rd_map.end();) {
+    for (auto it = m_rd_map.begin(); it != m_rd_map.end();) {
         auto rdev = std::dynamic_pointer_cast< RaftReplDev >(it->second);
         if (rdev->is_destroy_pending() &&
             (get_elapsed_time_sec(rdev->destroyed_time()) >=
```

Comment:
is there a reason why we need this check given this function is only called in a configured timer (`generic.repl_dev_cleanup_interval_sec`) already?

Reply:
The timer could wake up more frequently and it is a common timer for all repl_devs. Also each repl_dev would like to want to wait more time, especially in cases of migrate PG. So adding additional check to be more precise.

Reply:
Can we put a comment above the if by saying `// Although we have a global timer, we want precise control on each individual rdev. `, just for readability?